internal/dbutil/pgutil/schema.go

```diff
@@ -8,18 +8,46 @@ import (
 	"crypto/rand"
 	"database/sql"
 	"encoding/hex"
+	"io"
 	"net/url"
 	"strconv"
 	"strings"
 )
 
-// RandomString creates a random safe string
-func RandomString(n int) string {
+const (
+	randRetries = 2
+)
+
+// RandomStringFromReader creates a random safe string sourced from the specified reader.
+func RandomStringFromReader(n int, r io.Reader) string {
+	failed := false
 	data := make([]byte, n)
-	_, _ = rand.Read(data)
+	for i := 0; i < randRetries; i++ {
```

Comment:
If it failed once then it will probably fail all the other tries. It makes more sense to kill the program immediately when this happens, because there's something very wrong with the system in that case.